### PR TITLE
Kobun the breaddog

### PR DESCRIPTION
**Pull request by CursedBirb**
## About The Pull Request
Adds third kitchen craftable animal - Kobun the bread dog. Currently we have Keeki the cak and Terrygold the butter bear. Keeki is best friend of sec due to frosting and Terrygold seems like second clown with its slips, Kobun is designed as help to Psychologist due to powerfull mood aura. Mood granted by aura stays as long as you are within range of aura and when you step out is expires in 10 seconds. You can trade mood aura for a possibility of wearing Kobun as hat

<details><summary>Old sprites</summary>

https://user-images.githubusercontent.com/47064429/182676917-888a1a44-ee5a-459f-b460-a70153ce9473.mp4

https://user-images.githubusercontent.com/47064429/182676927-0acccfc9-cb1e-4fbf-8269-8a679a349f71.mp4

https://user-images.githubusercontent.com/47064429/182676931-cff8eec9-db3d-43f4-86a5-1f878035ceff.mp4

https://user-images.githubusercontent.com/47064429/182676935-4badc5c6-46d3-4a4b-bdba-cd79b0a8af25.mp4

</details>
<details><summary>New sprites</summary>

d4e1658f9069dcc948a55ba39489bd07
9175bde10757b9aab9806d522ed57034
473210facc03f4a57ae9a9ac48165ed0

</details>

## Why It's Good For The Game
We already have two crafting animals, one made out of cake and another out of butter so why not add another one made out bread? Kobun was made to help people with low mood by giving it mood aura that will bring them back onto their feets

## Changelog

:cl:
add: Kobun the bread dog that helps people with low mood
add: Adds crafting recipe for Kobun
/:cl:

### CODE REVIEW

**Comment by CursedBirb** [comment]
God i cleaned my fork and yet it still shows old commits

**Comment by CursedBirb** [comment]
Once https://github.com/tgstation/tgstation/pull/68763 will merged i will commit a fix to this one as Mothroach adds additional dead sprites for head wear

**Comment by Justice12354** [comment]
> Once #68763 will merged i will commit a fix to this one as Mothroach adds additional dead sprites for head wear

If you want me to, I can get you the folders without the mothroach dmis to make your job easier. I am adding the dead sprites for head wear bc else they'll be invisible on-head when dead

**Comment by san7890** [comment]
Can you put butter on the dog?

**Comment by CursedBirb** [comment]
> > Once #68763 will merged i will commit a fix to this one as Mothroach adds additional dead sprites for head wear
> 
> If you want me to, I can get you the folders without the mothroach dmis to make your job easier. I am adding the dead sprites for head wear bc else they'll be invisible on-head when dead

Nah. No need I will just copy your files and paste my sprites to the same file

**Comment by Colovorat** [comment]
I TELEPORTED BREAD

**Comment by CursedBirb** [comment]
Also to people who wanna ask why Sanallite is bigger when you wear it on the head. Sanallite is made out of fluffy bread which can take quite a squish. So when you hold it in hands you squish it from all sides but when you wear it on the head the dominant force on it is our mysterious invisible ceiling

**Comment by CursedBirb** [comment]
> Can you put butter on the dog?

Technically no. But if take butter and change it into a liquid form and then splash it on Sanallite it will be technically covered in butter

**Comment by jlsnow301** [comment]
What is the name in reference to? Is there anything more dog-like?

**Review by Imaginos16 — CHANGES_REQUESTED** [review]
The sprites related to the breaddog are not up to the standards we’ve been recently attempting to pursue. I'd ask for you to improve it please.

**Comment by CursedBirb** [comment]
Converted to draft till i make it according to our standards

**Comment by CursedBirb** [comment]
Btw. If we hit limit of active pr's just close my pr. I will just ask later for reopen

**Comment by CursedBirb** [comment]
Sprites has been updated. Now more /tg/ bread-like

**Comment by Twaticus** [comment]
i dont understand the purple electricity thing. what is that stuff on top of it? and the name too what is a sanallite

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -711,3 +711,59 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
...
+	deathsound = SFX_BODYFALL
+	held_state = "breaddog"
+	worn_slot_flags = ITEM_SLOT_HEAD
+
+/mob/living/simple_animal/pet/dog/breaddog/add_cell_sample()
+	return
+
+/mob/living/simple_animal/pet/dog/breaddog/CheckParts(list/parts)
+	..()
+	var/obj/item/organ/internal/brain/B = locate(/obj/item/organ/internal/brain) in contents
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:741]: wouldn't this be in the parts list? or is that not what it's for
also actual var name please, B means nothing

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -711,3 +711,59 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
...
+	if(!B || !B.brainmob || !B.brainmob.mind)
+		return
+	B.brainmob.mind.transfer_to(src)
+	to_chat(src, "<span class='big bold'>You are a bread dog!</span><b> You're a harmless dog/bread hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:745]: you should use the span defines

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 3]
```diff
@@ -711,3 +711,59 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
...
+	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
+	free bread to the station!</b><b>'I’m not alone, and you aren’t either'</b>")
+	var/new_name = tgui_input_text(src, "Enter your name, or press \"Cancel\" to stick with Sanallite.", "Name Change", max_length = MAX_NAME_LEN)
+	if(new_name)
+		to_chat(src, span_notice("Your name is now <b>\"new_name\"</b>!"))
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:750]: you need to actually embed the name with [], test your code 🗞️

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 4]
```diff
@@ -711,3 +711,59 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
...
+	..()
+	if(stat)
+		return
+
+	if(health < maxHealth)
+		adjustBruteLoss(-4 * delta_time) //Fast life regen
+
+	for(var/mob/living/L in view(3, src)) //Mood aura which stay as long you do not wear Sanallite as hat or carry(I will try to make it work with hat someday(obviously weaker than normal one))
+		if(iscarbon(L))
+			SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "sanallite", /datum/mood_event/sanallite)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:763]: why not var/mob/living/carbon then?

Comment (LemonInTheDark) [reply code/modules/mob/living/simple_animal/friendly/dog.dm:763]: if this range wasn't so small I'd be telling you to use a similar pattern to the eggborg's antiharm sphere

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
Do we really need another bakeable pet?

**Comment by LemonInTheDark** [comment]
> i dont understand the purple electricity thing. what is that stuff on top of it? and the name too what is a sanallite

did you ever work this out twat

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 3 above)
Comment (CursedBirb) [reply code/modules/mob/living/simple_animal/friendly/dog.dm:750]: This used already existing code for can and bear but I'm already testing []

**Review by Imaginos16 — APPROVED** [review]
The sprites are better now, but you should address what Twaticus said. And its up to other coders whether or not it gets added.

**Review by Twaticus — APPROVED** [review]
sprites cute, and i leared its a cosmic jelly which is interesting. the names a vtuber reference though so do with that what you will c*ders

**Comment by CursedBirb** [comment]
What the hell it appeared here. God i hate rebase

**Comment by CursedBirb** [comment]
Shit it eaten all my things

**Comment by CursedBirb** [comment]
That banana sprite that appreared for some reason due rebase was replaced by sprite i downloaded from github(for some reason it did not disappared from pr change)

**Comment by CursedBirb** [comment]
I want to cry now

**Comment by CursedBirb** [comment]
Im closing it for some time to fix my fork

**Comment by JohnFulpWillard** [comment]
> sprites cute, and i leared its a cosmic jelly which is interesting. the names a vtuber reference though so do with that what you will c*ders

wtf no vtuber reference why

**Comment by CursedBirb** [comment]
Name is the only direct reference so i will change it

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 5]
```diff
@@ -711,3 +711,57 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
...
+
+/mob/living/simple_animal/pet/dog/breaddog/CheckParts(list/parts)
+	..()
+	var/obj/item/organ/internal/brain/candidate = locate(/obj/item/organ/internal/brain) in contents
+	if(!candidate || !candidate.brainmob || !candidate.brainmob.mind)
+		return
+	candidate.brainmob.mind.transfer_to(src)
+	to_chat(src, "[span_boldbig("You are a bread dog!")]<b> You're a harmless dog/bread hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
+	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
+	free bread to the station!</b><b>'I’m not alone, and you aren’t either'</b>")
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:745]: indent these two lines with tab once, to make this easier to read

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 6]
```diff
@@ -711,3 +711,57 @@ GLOBAL_LIST_INIT(strippable_corgi_items, create_strippable_list(list(
...
+/mob/living/simple_animal/pet/dog/breaddog/Life(delta_time = SSMOBS_DT, times_fired)
+	..()
+	if(stat)
+		return
+
+	if(health < maxHealth)
+		adjustBruteLoss(-4 * delta_time) //Fast life regen
+
+	for(var/mob/living/carbon/L in view(3, src)) //Mood aura which stay as long you do not wear Sanallite as hat or carry(I will try to make it work with hat someday(obviously weaker than normal one))
+		L.add_mood_event("kobun", /datum/mood_event/kobun)
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/dog.dm:761]: is there a proper name for this that isn't just L? what is this carbon to this proc?

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 7]
```diff
@@ -295,16 +295,17 @@
...
 		return
-	B.brainmob.mind.transfer_to(src)
-	to_chat(src, "<span class='big bold'>You are a cak!</span><b> You're a harmless cat/cake hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
+	candidate.brainmob.mind.transfer_to(src)
+	to_chat(src, "[span_boldbig("You are a cak!")]<b> You're a harmless cat/cake hybrid that everyone loves. People can take bites out of you if they're hungry, but you regenerate health \
 	so quickly that it generally doesn't matter. You're remarkably resilient to any damage besides this and it's hard for you to really die at all. You should go around and bring happiness and \
 	free cake to the station!</b>")
-	var/new_name = tgui_input_text(src, "Enter your name, or press \"Cancel\" to stick with Keeki.", "Name Change", max_length = MAX_NAME_LEN)
+	var/default_name = "Keeki"
+	var/new_name = sanitize_name(reject_bad_text(tgui_input_text(src, "You are the [name]. Would you like to change your name to something else?", "Name change", default_name, MAX_NAME_LEN)))
```
Comment (LemonInTheDark) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:306]: pass in cap_after_symbols = FALSE for sanitize name, so thing's pet doesn't get turned into thing'S pet

**Pull request merged**